query-service-impl/src/main/java/org/hypertrace/core/query/service/attribubteexpression/AttributeExpressionNormalizationTransformation.java

```diff
@@ -0,0 +1,41 @@
+package org.hypertrace.core.query.service.attribubteexpression;
+
+import io.reactivex.rxjava3.core.Single;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.query.service.AbstractQueryTransformation;
+import org.hypertrace.core.query.service.QueryRequestUtil;
+import org.hypertrace.core.query.service.api.ColumnIdentifier;
+import org.hypertrace.core.query.service.api.Expression;
+import org.slf4j.Logger;
+
+@Slf4j
+final class AttributeExpressionNormalizationTransformation extends AbstractQueryTransformation {
+
+  @Override
+  public int getPriority() {
+    // Run before default transformations
+    return 1;
+  }
+
+  @Override
+  protected Logger getLogger() {
+    return log;
+  }
+
+  @Override
+  protected Single<Expression> transformColumnIdentifier(ColumnIdentifier columnIdentifier) {
+    Expression.Builder expressionBuilder =
+        QueryRequestUtil.createSimpleAttributeExpression(columnIdentifier.getColumnName());
+    if (columnIdentifier.getAlias().isBlank()) {
+      return Single.just(expressionBuilder.build());
+    }
+
+    return Single.just(
+        expressionBuilder
+            .setAttributeExpression(
```

Comment:
Don't we need to unset the present attribute expression first ?

Reply:
Not sure I follow. There is no attribute expression to start, we've created one inside an expression on line 27. Here, we update that attribute expression to include an alias.

Reply:
Yes, one created on line 27. Don't we need to unset that ?

Reply:
We're overwriting it - an expression can only contain one attribute expression, so we're reading it's value out, updating it and writing it back over the original.